### DISCUSSION TITLE
Explanation and relationship sorting of various indicators for Nervos DAO

### DISCUSSION THREAD

**Discussion by Keith-CY** [nervosnetwork/ckb#3939] [Q&A]
Hi there, we're going to sort out Nervos DAO-related logic/data in CKB Explorer and have made a sketch at https://docs.google.com/document/d/1BmRMycFFgPe67AJXs1z8VpqzMM3t7YtJAJTTwXFdrxg/

Among them, the equation at https://docs.google.com/document/d/1BmRMycFFgPe67AJXs1z8VpqzMM3t7YtJAJTTwXFdrxg/edit?disco=AAAAupRYkxY

```
mining_rewards + deposit_compensation + treasury == sum(block_economic_state.issuance.secondary) == header.dao.C_i
```

is confusing and needs confirmation from the core team.

The explanation of these fields can be found in the docs.

---

Also, it would be greatly helpful if you can have a review on the docs.

**Comment by doitian** [comment]
C_i is the total circulation amount, which should be equal to the sum of the live cells capacity when Block i is the head of the chain + S_i + the miner rewards of the ten blocks that have not yet been cast [i -9, i]. In the future, this equation may not hold, when the miner rewards are not enough to create a cell, this part of the miner rewards will not be cast directly. So the right side also has to add up the miner rewards that have not been cast because of this situation. 

C_i has the following destinations:
- The total number of live cells created in the genesis block, less than this part of the above equation
- The number issued to miners and successfully cast, now equal to the mining_rewards in the equation
- The number issued to miners but not cast because it is less than one Cell, now equal to 0, and will likely remain 0 for about 20 years
- Except for the part given to the miners, the other secondary rewards will be deposited in S_i
- When redeeming the DAO, the corresponding compensation will be obtained, which will be deducted from S_i

Among them, 1 + 2 + 5 - [i-9, i] The miner rewards of these ten blocks are equal to the total number of live cells.

C_i includes primary issuance and genesis block, and is definitely not equal to sum(block_economic_state.issuance.secondary). 

Part of sum(block_economic_state.issuance.secondary) has been issued to miners through cellbase, and the rest is first deposited in S_i and deducted from S_i when the DAO is redeemed. So 

sum(block_economic_state.issuance.secondary) = miner_rewards_l2 + Redeemed DAO compensation + S_i 

The amount of S_i belonging to the treasury cannot be known because the DAO calculation does not need it, and there is no RPC to return this information. An accurate calculation method is: Calculate the total amount of compensation required if all DAO cells are redeemed in block i, deduct this part from S_i, and the remaining is the treasury. Because the starting points are different, this calculation can only be calculated one by one and then summed up, and there is no recursion that can be used.

Since basically do not mention the treasury, it is recommended that the chart be divided into miner rewards, issued DAO compensation, and remaining secondary issuance not yet cast. Personally, I do not recommend using burn, one is that the treasury does not burn, just not cast, and the other is that it takes too much calculation to distinguish between S_i's treasury and unclaimed DAO compensation.

-- Translated via Claude

<details><summary>Original answer posted in Chinese</summary>

C_i 是流通总量，应该等于在 Block i 作为链头时的 live cells capacity 之和 + S_i + [i -9, i] 这十个块还没铸造出来的矿工奖励。在未来这个等式有可能不成立，当矿工奖励不够创建出一个 cell，这部分矿工奖励就直接不铸造了。所以右边还得加上所以因为这种情况没有铸造出来的矿工奖励。

C_i 的去向有以下几种:
创世块中创建出来的 live cells 总量，上面等式里少了这部分
发放给矿工并且成功铸造出来的数量，现在是等于等式中的 mining_rewards
发放给矿工但是因为不够一个 Cell 所以没有铸造出来的数量，现在等于 0，并且大概在 20 年内都会等于 0
除开给矿工的部分，所以其它二级奖励会存入 S_i
当赎回 DAO 时，会获得对应的补偿，这部分补偿从 S_i 里扣除

其中 1 + 2 + 5 - [i-9, i] 这十个块的矿工奖励等于 live cells 总量

C_i 是包含一级增发和创世块得，肯定不等于 sum(block_economic_state.issuance.secondary) 。sum(block_economic_state.issuance.secondary)  一部分是通过 cellbase 发给矿工了，剩下的都先存入 S_i，DAO 赎回再从 S_i 里扣。所以 

sum(block_economic_state.issuance.secondary) = miner_rewards_l2 + 已领取的 DAO 补偿 + S_i

其中 S_i 里有多少是 treasury 的这个因为 DAO 计算不需要，所以没办法知道。因为也没有 RPC 会返回这个信息。一个准确的计算方法是：

计算如果把所有 DAO 中的 cell 在 block i 中赎回，计算所需要的补偿总额，从 S_i 里扣除掉这部分剩下的就是 treasury 了。因为起存点不同，这个计算只能是逐个计算再求和，也没有递归公司可以用，而且我们也基本不提 treasury，建议图表就分成 miner rewards, issued DAO compensation，和剩余未铸造的二级增发。个人不建议用 burn，一是 treasury 也没有 burn 只是没铸造，二是要区分 S_i 中的 treasury 和还没领取的 DAO 补偿计算量太大。

</details>

**Comment by doitian** [comment]
I forgot a solution to calculate the unsettled DAO interest.

Since explorer has already stored the total locked CKB in DAO.

In block i, the secondary issuance for DAOi = $floor( s_i * D_{i-1} / C_{i-1} )$

where:

- `s_i` is secondary issurance for block i
- `D_{i_1}` is total locked DAO after block i - 1
- `C_{i_i}` is the field C in the field DAO of the block i - 1

If we sum up DAOi from genesis, we get the total DAO compensation. If we substract the issued DAO compensation, we get the unsttled compensation still in `S`.
